server/src/main/java/com/linecorp/centraldogma/server/internal/storage/project/ProjectApiManager.java

```diff
@@ -0,0 +1,128 @@
+    private final MetadataService metadataService;
+
+    public ProjectApiManager(ProjectManager projectManager, CommandExecutor commandExecutor,
+                             MetadataService metadataService) {
+        this.projectManager = projectManager;
+        this.commandExecutor = commandExecutor;
+        this.metadataService = metadataService;
+    }
+
+    public Map<String, Project> listProjects() {
+        final Map<String, Project> projects = projectManager.list();
+        if (isAdmin()) {
+            return projects;
+        }
+
+        return listProjectsWithoutDogma(projects);
+    }
+
+    private static boolean isAdmin() {
+        final User currentUserOrNull = AuthUtil.currentUserOrNull();
+        if (currentUserOrNull == null) {
+            return false;
+        }
+
+        return currentUserOrNull.isAdmin();
+    }
+
+    public static ImmutableMap<String, Project> listProjectsWithoutDogma(Map<String, Project> projects) {
+        final Map<String, Project> result = new LinkedHashMap<>(projects.size() - 1);
+        for (Map.Entry<String, Project> entry : projects.entrySet()) {
+            if (!INTERNAL_PROJECT_DOGMA.equals(entry.getKey())) {
+                result.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return ImmutableMap.copyOf(result);
+    }
+
+    public Map<String, Instant> listRemovedProjects() {
+        return projectManager.listRemoved();
+    }
+
+    public CompletableFuture<Void> createProject(String projectName, Author author) {
+        checkInternalDogmaProject(projectName, "create");
+        return commandExecutor.execute(Command.createProject(author, projectName));
+    }
+
+    private static void checkInternalDogmaProject(String projectName, String methodName) {
+        if (INTERNAL_PROJECT_DOGMA.equals(projectName)) {
+            throw new IllegalArgumentException("Cannot " + methodName + ' ' + projectName);
+        }
+    }
+
+    public CompletableFuture<ProjectMetadata> getProjectMetadata(String projectName) {
+        return metadataService.getProject(projectName);
+    }
+
+    public CompletableFuture<Void> removeProject(String projectName, Author author) {
+        checkInternalDogmaProject(projectName, "remove");
+        // Metadata must be updated first because it cannot be updated if the project is removed.
+        return metadataService.removeProject(author, projectName)
+                              .thenCompose(unused -> commandExecutor.execute(
+                                      Command.removeProject(author, projectName)));
+    }
+
+    public CompletableFuture<Void> purgeProject(String projectName, Author author) {
+        checkInternalDogmaProject(projectName, "purge");
+        return commandExecutor.execute(Command.purgeProject(author, projectName));
+    }
+
+    public CompletableFuture<Revision> unremoveProject(String projectName, Author author) {
+        checkInternalDogmaProject(projectName, "patch");
```

Comment:
Is `patch` used intentionally? Wrote a comment in case of typo.
```suggestion
        checkInternalDogmaProject(projectName, "unremove");
```